File: ui/src/app/dto/order/order-response-dto.ts
import { ShopItemResponseDto } from "../shoppingcart/shopitem-response-dto";


export class OrderResponseDto {

    id: number;
    total: number;
    shippingFee: number;
    vatRate: number;
    statusOrder: string;
    dateOrder: string;
    methodOfPayment: string;
    items: ShopItemResponseDto[]

    constructor(data?: Partial<OrderResponseDto>) {
        this.id = data?.id ?? 0;
        this.total = data?.total ?? 0;
        this.shippingFee = data?.shippingFee ?? 0;
        this.vatRate = data?.vatRate ?? 0;
        this.statusOrder = data?.statusOrder ?? '';
        this.dateOrder = data?.dateOrder ?? '';
        this.methodOfPayment = data?.methodOfPayment ?? '';
        this.items = data?.items?.map(item => new ShopItemResponseDto(item)) ?? [];
    }

}

File: api/Bookshop.Application/Features/Orders/OrderResponseDto.cs
using Bookshop.Application.Features.Dto;
using Bookshop.Application.Features.ShoppingCarts;

namespace Bookshop.Application.Features.Orders
{
    public class OrderResponseDto : IBaseDto
    {
        public long Id { get; set; }
        public decimal SubTotal { get; set; }
        public decimal Total { get; set; }
        public decimal ShippingFee { get; set; }
        public decimal VatRate { get; set; }
        public string? StatusOrder { get; set; }
        public string? DateOrder { get; set; }
        public string? MethodOfPayment { get; set; }
        public IList<ShopItemResponseDto>? Items { get; set; }
    }
}
